app/lib/analyzer/backend.dart

```diff
@@ -63,10 +63,7 @@ class AnalysisBackend {
   /// Stores the analysis, and either creates or updates its parent
   /// [PackageAnalysis] and [PackageVersionAnalysis] records.
   Future storeAnalysis(Analysis analysis) async {
-    assert(analysis.id == null);
-    await db.commit(inserts: [analysis]);
-    assert(analysis.id != null);
-
+    analysis.id ??= new DateTime.now().millisecondsSinceEpoch;
```

Comment:
Not sure which API doc you are referring to. Of course the int can be anything. But if we use timestamps, we should make them utc.

There is also a way to ask the datastore to generate uniqe ids using [dbService.allocateIds()](https://www.dartdocs.org/documentation/gcloud/0.4.0/gcloud.datastore/Datastore/allocateIds.html). This way we would not even risk having low probability of clash (as long as all `Analysis` objects use `allocateIds` then it's guaranteed to be unique).

If you think it would be cleaner, then let's do it like this, otherwise let me know and I'll merge it as it is now.